src/main/service.rs

```diff
@@ -75,7 +75,19 @@ impl<UID: Uid> Service<UID> {
         let mut mc = MappingContext::new()?;
         mc.add_peer_stuns(config.hard_coded_contacts.iter().cloned());
 
-        let el = common::spawn_event_loop(4, Some(&format!("{:?}", our_uid)))?;
+        let bootstrap_cache_file = config.bootstrap_cache_name.clone();
+        let el = common::spawn_event_loop(4, Some(&format!("{:?}", our_uid)), move || {
+            match BootstrapCache::new(bootstrap_cache_file.as_ref()) {
+                Ok(cache) => {
+                    cache.read_file();
+                    Some(cache)
+                }
+                Err(e) => {
+                    error!("Failed to initialize bootstrap cache: {}", e);
+                    None
```

Comment:
Instead of an optional, create an in-mem file otherwise you are coupling `T` in event loop module too much with the implementation here. If even in-mem etc are not possible then it's better to error out here than check for a valid optional in event-loop module (`common/core.rs`) and error out there.